sdk/python/kfp/components/type_annotation_utils.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for handling Python type annotation."""
+
+import re
+from typing import TypeVar, Union
```

Comment:
Is this a public module intended to be a part of the public KFP API?

Reply:
No, it's not intended as a public API. 

That said, I'm a bit confused about the naming conventions here. Aren't all modules "private" in KFP SDK, and then we chose what to expose as public using the package level `__init__.py`. I got this impression because I don't expect users to import anything from a specific module. e.g. KFP users should always use `dsl.PipelineParam` but not `dsl._pipeline_param.PipelineParam`.
And if all modules are private, should we just drop the underscore prefix on all files instead of prefix every single file?